### ISSUE TITLE
conda lock files missing `pyarrow-tests`

### ISSUE THREAD

**Issue by ncclementi** [ibis-project/ibis#7383]
After I went through the env setup here https://ibis-project.org/contribute/01_environment following the `mamba` option with the conda lock. 
I tried to run the tests but run into 
```python-traceback
___________________ ERROR collecting ibis/formats/tests/test_pyarrow.py ____________________
ImportError while importing test module '/Users/naty/Documents/git/my_forks/ibis/ibis/formats/tests/test_pyarrow.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/Users/naty/mambaforge/envs/ibis-dev/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ibis/formats/tests/test_pyarrow.py:6: in <module>
    import pyarrow.tests.strategies as past
E   ModuleNotFoundError: No module named 'pyarrow.tests'

```

This gets resolved by doing (Thanks @gforsyth )
```
mamba install pyarrow-tests
```

So it'll probably be good to have it in the conda lock file.

**Comment by gforsyth** [comment]
Hmm, I think the (interesting) root cause of this is that the `pyarrow` wheel distributed on PyPI contains the `tests` submodule, but the conda-forge `pyarrow` package does not.

**Comment by cpcloud** [comment]
Fixed by #7390.
